craft_parts/sources/tar_source.py

```diff
@@ -74,20 +73,15 @@ def __init__(
         if source_depth:
             raise errors.InvalidSourceOption(source_type="tar", option="source-depth")
 
-    # pylint: enable=too-many-arguments
-
     @overrides
     def provision(
         self,
         dst: Path,
-        keep: bool = False,
-        src: Optional[Path] = None,
+        keep: bool = False,  # noqa: FBT001, FBT002
```

Comment:
I think we should make `keep` and `src` keyword arguments (sources are not supposed to be used directly by applications so it should be safe)

Reply:
It already keyword arguments, unless I misunderstand the word

Reply:
I think he meant mandatory kwargs, e.g.

```python
    def provision(
        self,
        dst: Path,
        *,
        keep: bool = False,
        src: Optional[Path] = None,
```